src/main/java/com/scalar/cassy/transferer/AzureBlobFileDownloader.java

```diff
@@ -1,63 +1,88 @@
 package com.scalar.cassy.transferer;
 
-import com.azure.storage.blob.BlobContainerAsyncClient;
+import com.azure.storage.blob.BlobContainerClient;
 import com.azure.storage.blob.models.BlobItem;
-import com.azure.storage.blob.models.BlobProperties;
 import com.azure.storage.blob.models.ListBlobsOptions;
+import com.google.common.annotations.VisibleForTesting;
 import com.google.inject.Inject;
 import com.scalar.cassy.config.RestoreConfig;
 import com.scalar.cassy.exception.FileTransferException;
-import java.io.IOException;
+import java.io.*;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
+import java.util.Iterator;
 import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import reactor.core.publisher.Mono;
 
 public class AzureBlobFileDownloader implements FileDownloader {
   private static final Logger logger = LoggerFactory.getLogger(AzureBlobFileDownloader.class);
-  private BlobContainerAsyncClient blobContainerClient;
+  private BlobContainerClient blobContainerClient;
+  private static final int NUM_THREADS = 3;
+  private final ExecutorService executorService = Executors.newFixedThreadPool(NUM_THREADS);
 
   @Inject
-  public AzureBlobFileDownloader(BlobContainerAsyncClient blobContainerClient) {
+  public AzureBlobFileDownloader(BlobContainerClient blobContainerClient) {
     this.blobContainerClient = blobContainerClient;
   }
 
   @Override
   public void download(RestoreConfig config) {
+    List<Future> downloadFuture = new ArrayList<>();
     String key = BackupPath.create(config, config.getKeyspace());
 
     logger.info("Downloading " + blobContainerClient.getBlobContainerName() + "/" + key);
-    List<Mono<BlobProperties>> filesToBeDownloaded = new ArrayList<>();
-    Iterable<BlobItem> keyspaceBlobs =
-        blobContainerClient.listBlobs(new ListBlobsOptions().setPrefix(key + "/")).toIterable();
-    for (BlobItem blob : keyspaceBlobs) {
+    Iterator<BlobItem> keyspaceBlobs =
```

Comment:
This change is required?
I know we tested if using Iterator makes things different, but it didn't, right?

Reply:
> This change is required?

Not required, 
But now we use synchronous `BlobContainerClient` instead of the asynchronous client.
Synchronous listBlobs will return an iterator object.

Need to revert to the previous implementation?

Reply:
Sorry, let's use `for` as previously.